### PR TITLE
enable multi root DnD

### LINKED ISSUE #44554: Allow to DND multiple root-folders
Testing #29715

I see no reason why we would not support to DND multiple root folders. The API is easy enough to use, just splice the folders by removing all and then adding all in the right order. We do the diff and will only send the correct change event, you don't have to compute it.

### PR DESCRIPTION
**Pull request by isidorn**
fixes #44554

Not the nicest code but gets the job done for multi select root DnD.
@bpasero please review and let me know if you know how to simplify this. Thanks!

### CODE REVIEW

**Comment by bpasero** [comment]
@isidorn thanks will do

File: src/vs/workbench/parts/files/electron-browser/views/explorerViewer.ts [hunk 1]
```diff
@@ -817,7 +817,7 @@ export class FileDragAndDrop extends SimpleFileResourceDragAndDrop {
 					return true; // NewStatPlaceholders can not be moved
 				}
 
-				if (source.isRoot && (sources.length > 1 || target instanceof FileStat && !target.isRoot)) {
+				if (source.isRoot && target instanceof FileStat && !target.isRoot) {
 					return true; // Root folder can not be moved to a non root file stat. Do not allow root folder move when multi selection drag.
```
Comment (bpasero) [inline src/vs/workbench/parts/files/electron-browser/views/explorerViewer.ts]: The comment needs an update too I think

File: src/vs/workbench/parts/files/electron-browser/views/explorerViewer.ts [hunk 2]
```diff
@@ -949,43 +949,49 @@ export class FileDragAndDrop extends SimpleFileResourceDragAndDrop {
...
-					}
-					if (folders[index].uri.toString() === targetUri.toString()) {
-						targetIndex = index;
-					}
-					workspaceCreationData.push({
-						name: folders[index].name,
-						uri: folders[index].uri
-					});
-				}
+	private doHandleRootDrop(roots: FileStat[], target: FileStat | Model): TPromise<void> {
```
Comment (bpasero) [inline src/vs/workbench/parts/files/electron-browser/views/explorerViewer.ts]: Wouldn't it be much easier here to do something like:
* fill an array of roots that includes everything except the roots that are being dragged
* find the index where to drop to (if target is model, the index is the length of the array)
* add the roots that are being dragged into the array at the correct location
* call workspace editing service with that array

**Review by bpasero — CHANGES_REQUESTED** [review]

File: src/vs/workbench/parts/files/electron-browser/views/explorerViewer.ts [hunk 3]
```diff
@@ -949,43 +949,49 @@ export class FileDragAndDrop extends SimpleFileResourceDragAndDrop {
...
+		const folders = this.contextService.getWorkspace().folders;
+		let targetIndex: number;
+		const workspaceCreationData: IWorkspaceFolderCreationData[] = [];
+		const targetUri = target instanceof FileStat ? target.resource : folders[folders.length - 1].uri;
+		const rootsToMove: IWorkspaceFolderCreationData[] = [];
+
+		for (let index = 0; index < folders.length; index++) {
+			const data = {
+				name: folders[index].name,
+				uri: folders[index].uri
```
Comment (bpasero) [inline src/vs/workbench/parts/files/electron-browser/views/explorerViewer.ts]: @isidorn also careful here, you are introducing a `` property that previously might not have been there

Comment (isidorn) [reply src/vs/workbench/parts/files/electron-browser/views/explorerViewer.ts]: @bpasero how can I check if the name was there previously?

File: src/vs/workbench/parts/files/electron-browser/views/explorerViewer.ts (hunk 2 above)
Comment (isidorn) [reply src/vs/workbench/parts/files/electron-browser/views/explorerViewer.ts]: Thanks for the suggestion. This simplifies a bit. This was tackled via 5564db9710

File: src/vs/workbench/parts/files/electron-browser/views/explorerViewer.ts (hunk 3 above)
Comment (bpasero) [reply src/vs/workbench/parts/files/electron-browser/views/explorerViewer.ts]: @isidorn you don't need to, just leave `name` empty

**Comment by isidorn** [comment]
Great, thanks for feedback. I feel like I have addressed all the issues -> merging

**Pull request merged**